### PR TITLE
update docs and fix flaky tests

### PR DESCRIPTION
**Pull request by mangalaman93**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,14 @@
 # pydgraph
 
-This is the official Dgraph database client implementation for Python (Python >= v2.7 and >= v3.5), using [gRPC][grpc].
+This is the official Dgraph database client implementation for Python (>= v3.5), using [gRPC][grpc].
```
Comment (joshua-goldstein) [inline README.md:3]: Let's say >=3.11, since this is what is in the `pyproject.toml` file, and what we are testing on Ci.

Comment (mangalaman93) [reply README.md:3]: Done.

File: tests/test_txn.py [hunk 2]
```diff
@@ -234,8 +235,11 @@ def test_read_only_txn(self):
         """Tests read-only transactions. Read-only transactions should
         not advance the start ts nor should allow mutations or commits."""
 
-        query = '{ me() {} }'
+        # We sleep here so that rollups do not move the MaxAssigned.
+        # Starting Dgraph v23, rollups can move the MaxAssigned too.
+        time.sleep(1)
```
Comment (joshua-goldstein) [inline tests/test_txn.py:240]: Do we want to add a link to https://github.com/dgraph-io/dgraph/pull/8774?

Comment (mangalaman93) [reply tests/test_txn.py:241]: Done.

**Review by joshua-goldstein — APPROVED** [review]

**Pull request merged**